### PR TITLE
crimson/os/seastore/cache: consider EXIST_CLEAN extents as pending ones

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Signed-off-by: Xuehan Xu <xxhdx1985126@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/os/seastore/cached_extent.h [hunk 1]
```diff
@@ -284,7 +284,8 @@ class CachedExtent : public boost::intrusive_ref_counter<
   bool is_pending() const {
```
Comment (cyx1231st) [inline src/crimson/os/seastore/cached_extent.h:284]: How about splitting into the following two methods:

```c++
/// Returns true if extent can be mutated in an open transaction
bool is_mutable() const {
  return state == extent_state_t::INITIAL_WRITE_PENDING ||
      state == extent_state_t::MUTATION_PENDING ||
      state == extent_state_t::EXIST_MUTATION_PENDING;
}

/// Returns true if extent is part of an open transaction
bool is_pending() const {
  return is_mutable() || state == extent_state_t::EXIST_CLEAN;
}
```

The two methods imply different usage scenarios:
* Most of the existing users related to the `Cache::duplicate_for_write()` path should check `is_mutable()` for the decision.
* LBA-ptrs checks the new `is_pending()` to identify if it is part of a transaction.

File: src/crimson/os/seastore/cache.cc [hunk 2]
```diff
@@ -999,9 +999,6 @@ CachedExtentRef Cache::duplicate_for_write(
   Transaction &t,
   CachedExtentRef i) {
   LOG_PREFIX(Cache::duplicate_for_write);
-  if (i->is_pending())
```
Comment (cyx1231st) [inline src/crimson/os/seastore/cache.cc:1002]:
```c++
if (i->is_mutable()) {
  return i;
}
```

Comment (xxhdx1985126) [reply src/crimson/os/seastore/cache.cc:1002]: done.

**Comment by cyx1231st** [comment]
@xxhdx1985126 Does the commit https://github.com/ceph/ceph/commit/dde49af7d01d4512fa78717657dbdb71a4557f60 reasonable to you?

**Comment by xxhdx1985126** [comment]
> @xxhdx1985126 Does the commit [dde49af](https://github.com/ceph/ceph/commit/dde49af7d01d4512fa78717657dbdb71a4557f60) reasonable to you?

Ah, yes, I'll cherry-pick that commit

**Comment by xxhdx1985126** [comment]
@cyx1231st PLTK, thanks:-)

**Review by cyx1231st — APPROVED** [review]

**Pull request merged**